packages/sdk/src/tables/base/table.ts

```diff
@@ -2,7 +2,10 @@ import assert from "assert";
 import { exists } from "../../utils";
 import type { stores, MakeId, MaybeId, HasId } from "../..";
 
-export default function Table<I, D>(config: { makeId: MakeId<I, D>; type: string }, store: stores.Store<I, D>) {
+export default function Table<I, D, S extends stores.Store<I, D>>(
```

Comment:
`S extends stores.Store<I, D>`
I use Generics for the store because I want the returned type to specify only the supported methods by the Store. 
`Store vs SortedStore`: `SortedStore` has additional functions: `slice()` and `between()`

Reply:
i considered this previously, but maybe we can just have a single store interface where a non-sorted store either throws an error or returns an empty array for unsupported calls